modules/instrument_builder/jsx/react.questions.js

```diff
@@ -394,13 +394,6 @@ class DateOptions extends Component {
     this.onChange = this.onChange.bind(this);
   }
 
-  /**
-   * Called by React when the component has been rendered on the page.
-   */
-  componentDidMount() {
-    this.props.element.Options.dateFormat = '';
```

Comment:
Unsure why this line existed and had to be removed for solving the dropdown of type of date not being set correctly when editing the date field.